aiida/orm/node/process/calculation/calcjob.py

```diff
@@ -443,47 +192,6 @@ def _validate(self):
         except (TypeError, ValueError) as exc:
             raise ValidationError("Invalid resources for the scheduler of the specified computer: {}".format(exc))
 
-        if not isinstance(self.get_option('withmpi', only_actually_set=False), bool):
-            raise ValidationError("withmpi property must be boolean! It in instead {}"
-                                  "".format(str(type(self.get_option('withmpi')))))
-
-    def validate_incoming(self, source, link_type, link_label):
-        """
-        Validate adding a link of the given type from a given node to ourself.
-
-        :param source: the node from which the link is coming
-        :param link_type: the type of link
-        :param link_label: link label
-        :raise TypeError: if `source` is not a Node instance or `link_type` is not a `LinkType` enum
-        :raise ValueError: if the proposed link is invalid
-        """
-        super(CalcJobNode, self).validate_incoming(source, link_type, link_label)
-
-        state = self.get_state()
-        valid_states = [calc_states.NEW]
-        if state not in valid_states:
-            raise ModificationNotAllowed('invalid link: CalcJobNode has to have state in {}, but is {}'.format(
-                valid_states, state))
-
-    def validate_outgoing(self, target, link_type, link_label):
```

Comment:
Don't we want to still do some validation depending on the (process_)state? Or it's somewhere else?